frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -33,27 +78,28 @@ const AnnouncementModal = () => {
     }
   }
 
+  // Render the appropriate modal based on the type of content
+  const translations = getTranslations()
+  let specificAnnouncementModal = null
+  if (isVideoUrl(translations.mediaUrl!)) {
```

Comment:
```suggestion
  if (translation.mediaUrl && isVideoUrl(translations.mediaUrl)) {
```

If `mediaUrl` is `undefined`, `isVideoUrl` will throw an error at runtime. Generally don't ignore warnings from eslint unless you have a good reason. Use the types to help avoid runtime error.

Reply:
Thanks! I was unsure about this but just assumed that it would never happen. Now that you mention it, such an assumption isn't good to have. I've strengthened the type definitions and removed the implicit assumptions of non-null-ness.